### PR TITLE
Rachaelbond patch 3

### PR DESCRIPTION
**Pull request by rachaelbond**
Hopefully no linting errors now! Note to self: don't sync a forked branch that you're working on with the master :-/

### CODE REVIEW

**Comment by Anrolosia** [comment]
> @Anrolosia I have no idea why the lint checker is kicking up about config_flow.py :(

I'll take a look at your PR and I will fix the linting once merged

**Comment by rachaelbond** [comment]
@Anrolosia the last version of config_flow.py *should* be correct. Can't see a problem with it...

File: custom_components/shopping_list_with_grocy/__init__.py [hunk 1]
```diff
@@ -129,6 +129,31 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 
     return True
 
+    #
+    # To v4
+    #
+
+    if config_entry.version == 3:
+        v2_options = dict(config_entry.options)
+        v2_data = dict(config_entry.data)
```
Comment (Anrolosia) [inline custom_components/shopping_list_with_grocy/__init__.py:138]: Could you use same syntax as above for variable initialization please?

File: custom_components/shopping_list_with_grocy/__init__.py [hunk 2]
```diff
@@ -129,6 +129,31 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
...
+        if v2_options('adding_images', FALSE) is TRUE:
+            v2_options['image_download_size'] = 100
+            v2_data['image_download_size'] = 100
+        else:
+            v2_options['image_download_size'] = 0
+            v2_data['image_download_size'] = 0
+
+        v2_options.pop('adding_images', NONE)
```
Comment (Anrolosia) [inline custom_components/shopping_list_with_grocy/__init__.py:146]: if `adding_images` also exists in v2_data, it should also be removed from there (options are when config is changed AFTER plugin init, data if for options selected during plugin init)

File: custom_components/shopping_list_with_grocy/__init__.py [hunk 3]
```diff
@@ -129,6 +129,31 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
...
     return True
 
+    #
+    # To v4
+    #
+
+    if config_entry.version == 3:
+        v2_options = dict(config_entry.options)
+        v2_data = dict(config_entry.data)
+        if v2_options('adding_images', FALSE) is TRUE:
```
[the remark this answers is not in the document]
Comment (rachaelbond) [reply custom_components/shopping_list_with_grocy/__init__.py]: done

File: custom_components/shopping_list_with_grocy/__init__.py (hunk 1 above)
Comment (rachaelbond) [reply custom_components/shopping_list_with_grocy/__init__.py]: done

File: custom_components/shopping_list_with_grocy/__init__.py (hunk 2 above)
Comment (rachaelbond) [reply custom_components/shopping_list_with_grocy/__init__.py:146]: done

File: custom_components/shopping_list_with_grocy/apis/shopping_list_with_grocy.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def __init__(self, websession: ClientSession, hass: HomeAssistant, config):
         self.mqtt_port = config.get("mqtt_port", 1883)
         self.mqtt_username = config.get("mqtt_username", None)
         self.mqtt_password = config.get("mqtt_password", None)
-        self.fetch_images = config.get("adding_images", True)
+        self.fetch_images = config.get("image_download_size", 0)
```
[the remark this answers is not in the document]
Comment (rachaelbond) [reply custom_components/shopping_list_with_grocy/apis/shopping_list_with_grocy.py]: done "fetch_images" -> "image_size"

**Comment by Anrolosia** [comment]
Thanks for your contribution, I'll test all of that and create a release soon if everything is good !

**Comment by rachaelbond** [comment]
@Anrolosia no worries! sorry that was harder work than it should have been! I don't really code in python (C++, PHP no worries!) and I've never contributed to a Home Assistant integration before. I also loathe doing PRs through the web interface ;)

**Pull request merged**